### PR TITLE
Reenable zlib on 9x 

### LINKED ISSUE #88: Build llvmdev with zlib support
Issue:
If we try to use conda-forge clang-9 with `-flto -fuse-ld=ld.lld` we encounter the following error:

```
ld.lld: error: /work/sparta/conda/miniforge3/envs/dabble_env/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/libc_nonshared.a(elf-init.oS): contains a compressed section, but zlib is not available
```
<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
 packages in environment at /work/sparta/conda/miniforge3/envs/dabble_env:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       0_gnu    conda-forge
binutils_impl_linux-64    2.34                 h53a641e_0    conda-forge
binutils_linux-64         2.34                hc952b39_18    conda-forge
boost-cpp                 1.70.0               h8e57a91_2    conda-forge
bzip2                     1.0.8                h516909a_2    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
cairo                     1.16.0            hcf35c78_1003    conda-forge
certifi                   2020.4.5.1       py38h32f6830_0    conda-forge
clang                     9.0.1           default_hde54327_0    conda-forge
clangxx                   9.0.1           default_hde54327_0    conda-forge
cmake                     3.16.5               h28c56e5_0    conda-forge
curl                      7.68.0               hf8cf82a_0    conda-forge
doxygen                   1.8.16               hd1b7508_1    conda-forge
expat                     2.2.9                he1b5a44_2    conda-forge
fontconfig                2.13.1            h86ecdb6_1001    conda-forge
freetype                  2.10.1               he06d7ca_0    conda-forge
gcc_impl_linux-64         7.3.0                hd420e75_5    conda-forge
gcc_linux-64              7.3.0               h553295d_18    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
git                       2.26.0          pl526hf241897_0    conda-forge
glib                      2.58.3          py38h73cb85d_1003    conda-forge
gmp                       6.2.0                he1b5a44_2    conda-forge
graphite2                 1.3.13            he1b5a44_1001    conda-forge
gxx_impl_linux-64         7.3.0                hdf63c60_5    conda-forge
gxx_linux-64              7.3.0               h553295d_18    conda-forge
harfbuzz                  2.4.0                h9f30f68_3    conda-forge
hdf5                      1.10.6          nompi_h3c11f04_100    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
isl                       0.21                 he80fd80_1    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
krb5                      1.16.4               h2fd8d38_0    conda-forge
ld_impl_linux-64          2.34                 h53a641e_0    conda-forge
libcurl                   7.68.0               hda55be3_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.2.0                h24d8f2e_2    conda-forge
libgfortran-ng            7.3.0                hdf63c60_5    conda-forge
libgomp                   9.2.0                h24d8f2e_2    conda-forge
libiconv                  1.15              h516909a_1006    conda-forge
libllvm9                  9.0.1                hc9558a2_0    conda-forge
libpng                    1.6.37               hed695b0_1    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.2.0                hdf63c60_2    conda-forge
libtiff                   4.1.0                hc7e4089_6    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libuv                     1.34.0               h516909a_0    conda-forge
libwebp-base              1.1.0                h516909a_3    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.10               hee79883_0    conda-forge
lld                       9.0.1                h2295124_0    conda-forge
llvm-tools                9.0.1                hc9558a2_0    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
make                      4.3                  h516909a_0    conda-forge
mpc                       1.1.0             h04dde30_1007    conda-forge
mpfr                      4.0.2                he80fd80_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
openjpeg                  2.3.1                h981e76c_3    conda-forge
openssl                   1.1.1f               h516909a_0    conda-forge
pcre                      8.44                 he1b5a44_0    conda-forge
perl                      5.26.2            h516909a_1006    conda-forge
pip                       20.0.2                     py_2    conda-forge
pixman                    0.38.0            h516909a_1003    conda-forge
poppler                   0.65.0               h14e79db_0    conda-forge
poppler-data              0.4.9                         1    conda-forge
popt                      1.16              h299ea2f_2002    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
python                    3.8.2           h8356626_5_cpython    conda-forge
python_abi                3.8                      1_cp38    conda-forge
rapidjson                 1.1.0             he1b5a44_1002    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
rhash                     1.3.6             h14c3975_1001    conda-forge
riscv-gnu-toolchain       9.2.0                         0    bdutrosv
rsync                     3.1.3             hed695b0_1002    conda-forge
setuptools                46.1.3           py38h32f6830_0    conda-forge
sqlite                    3.30.1               hcee41ef_0    conda-forge
texlive-core              20180414        pl526h89d1741_1    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.9                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.5                h516909a_0    conda-forge
yaml-cpp                  0.6.3                he1b5a44_3    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
zstd                      1.4.4                h3b9ef0a_2    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : dabble_env
    active env location : /work/sparta/conda/miniforge3/envs/dabble_env
            shell level : 2
       user config file : /home/tims/.condarc
 populated config files : /work/sparta/conda/miniforge3/.condarc
                          /home/tims/.condarc
          conda version : 4.8.3
    conda-build version : 3.19.2
         python version : 3.7.6.final.0
       virtual packages : __glibc=2.23
       base environment : /work/sparta/conda/miniforge3  (read only)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /work/sparta/conda/miniforge3/pkgs
                          /home/tims/.conda/pkgs
       envs directories : /home/tims/.conda/envs
                          /work/sparta/conda/miniforge3/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.6 Linux/4.4.0-171-generic ubuntu/16.04.6 glibc/2.23
                UID:GID : 1404:1404
             netrc file : None
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by timsnyder**
Checklist
* [X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Fixes #88

### CODE REVIEW

**Comment by isuruf** [comment]
Do you need `9.x` or is master okay?

**Comment by timsnyder** [comment]
We probably need 9.x.  I know that we have some code that doesn't compile cleanly with 10.0 yet and I don't want to force that right now.

**Comment by timsnyder** [comment]
I'll also do this to master

**Comment by isuruf** [comment]
See warnings on linux,
```
2020-04-17T01:12:34.1663729Z WARNING (libllvm9,lib/libLLVM-9.so): $RPATH/libz.so.1 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
```
```
WARNING (llvmdev,libexec/llvm/lli-child-target): $RPATH/libz.so.1 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
```

**Comment by timsnyder** [comment]
@isuruf do you understand what they mean by 'sysroot' in that warning?  Do they mean the toolchain sysroot for gcc in `$PREFIX/x86_64_cos6-linux-gnu/`?  I killed my local build because docker desktop is slower than molasses on my macbook.  I'll go back and finish the build locally and try to understand where the RPATH is coming from.  I would expect it to be `$PREFIX/lib`.  Looking [above a warning that looks like the one you mentioned](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=145511&view=logs&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c&l=19597) I see that it looks like `$BUILD_PREFIX/lib` is being used as an RPATH in the llvm-tools and we definitely do not want that.  It's probably because `meta.yaml` puts `libllvm` in the `build` requirements instead of putting it in `host`?
```
19487 WARNING :: get_rpaths_raw()=['/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_build_env/lib:/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib:$ORIGIN/../lib'] and patchelf=[['/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_build_env/lib', '/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib', '$ORIGIN/../lib']] disagree for /home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/llvm-jitlink ::
19597 WARNING (llvm-tools,bin/llvm-jitlink): $RPATH/libLLVM-9.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
```


Note: looking back at [a CI build from Jan](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107597&view=logs&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c&l=18408) I also see:
```
Many lines talking about removing $BUILD_PREFIX/lib as an RPATH but looks like older conda-build didn't print the diagnostic line that included the offending binary or where the RPATH was being removed.

18408 WARNING (llvm-tools,bin/llvm-dis): $RPATH/libLLVM-9.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?

```

Even though these warnings about the rpath for `libLLVM09.so` aren't new with this PR, I agree that we should clean them up.  **Do you know why these are 'warnings' instead of being errors?**

I also need to fix the Winders build.  Looks like Windows isn't getting build and host environments for some reason and zlib isn't being found.

**Comment by isuruf** [comment]
You need to add `zlib` to `host` of all the outputs. Not just top level

**Comment by timsnyder** [comment]
> You need to add `zlib` to `host` of all the outputs. Not just top level

👍 . Even the `llvmdev` metapackage output?  I think everything looks to get the zlib support via `libllvm`.  I wonder if adding it only there would make the most sense?  I'll add it everywhere but expect that I'll get complaints about overdepending.

**Comment by timsnyder** [comment]
I guess `llvmdev` isn't really a metapackage though because it has install scripts and it does a `make install` without removing any of the components.  `libllvm{{major_ver}}` contains only the `lib/libllvm-{{major_ver}}.{so|dylib}` files and `llvmdev` contains the `lib/libllvm.{so|dylib}` symlinks that point to the major versions.

Oh, and I forgot about `run_exports` not being transitive.  At least I saw a comment about that someplace.  So, we need to add it to the host section everywhere.  Thanks for the suggestion @isuruf.  Sorry for questioning it.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -150,6 +154,7 @@ outputs:
       host:
         - python >=3
         - pip
+        - zlib
```
Comment (isuruf) [inline recipe/meta.yaml:157]: You don't need it here

Comment (timsnyder) [reply recipe/meta.yaml]: Ok.  Removed.

**Comment by timsnyder** [comment]
I'm still digging more into the [not found in sysroot warning](https://github.com/conda/conda-build/blob/d40cdabe31acb3515531167eb3edca36df9eee13/conda_build/post.py#L851). Trying to determine if this is a problem with pyldd or something real.  I think it is bogus because the RPATHS printed out on line 19487 (mentioned above) contain `$ORIGIN/../lib` and that should find libLLVM-9.so:
```
19487 WARNING :: get_rpaths_raw()=['/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_build_env/lib:/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib:$ORIGIN/../lib'] and patchelf=[['/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_build_env/lib', '/home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib', '$ORIGIN/../lib']] disagree for /home/conda/feedstock_root/build_artifacts/llvm-package_1587079994373/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/llvm-jitlink ::
```

**Comment by timsnyder** [comment]
@isuruf after sifting around in [conda_build/post.py](https://github.com/conda/conda-build/blob/d40cdabe31acb3515531167eb3edca36df9eee13/conda_build/post.py#L851) for a bit, it seems like `pylld` is not identifying `libllvm9` as the package that provides `libLLVM-9.so` and I don't think the Warnings being printed mean anything.  This seems like a bug in conda-build.  If I have time, I can try to root cause it and submit a PR on conda-build.

I'm inclined to merge this PR.  What do you think?

**Comment by isuruf** [comment]
Can you send a PR for master as well?

**Pull request merged**